Update devicons to v2.17.0

joeriddles opened the pull request:
#### Description

Devicons released [v2.17.0](https://github.com/devicons/devicon/releases/tag/v2.17.0) two days ago, which introduced many new icons. This PR upgrades devicons from v2.16.0 to v2.17.0.

#### Requirements / Checklist

- [x] Read the [Contributing Guidelines](https://github.com/ryanoasis/nerd-fonts/blob/-/contributing.md)
- [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan.
- [x] ~~If this contains a font/glyph add its origin as background info below (e.g. URL)~~ N/A
- [x] Verified the license of any newly added font, glyph, or glyph set. License is: xxx **Devicons uses MIT**

#### What does this Pull Request (PR) do?

- This PR upgrades devicons from v2.16.0 to v2.17.0.
- Updates the mapping file to use `original` instead of `plain` in cases were the `plain` file was removed/renamed in v2.17.0.
    - For example, appwrite was renamed to original [here](https://github.com/devicons/devicon/pull/2196/files#diff-8fb0089a81b777eae04d6ac8f37becd0a07c41dd808f17e9e90eafb7411a8f60)

#### How should this be manually tested?

Download the `devicons.otf` and inspect the new icons in your favorite font app.

#### Any background context you can provide?

#### What are the relevant tickets (if any)?
https://github.com/ryanoasis/nerd-fonts/issues/1899

#### Screenshots (if appropriate or helpful)
New icons, starting at the third one in from the top:
<img width="2110" height="3832" alt="image" src="https://github.com/user-attachments/assets/bc5b7ac4-1cf1-4f90-82fa-556c36758e7e" />

joeriddles committed:
Update devicons to v2.17.0

joeriddles committed:
Update devicons to v2.17.0

joeriddles committed:
Commit devicons.otf

and increase font version to be able to distinguish it.

$ fontforge ./generate
Copyright (c) 2000-2026. See AUTHORS for Contributors.
 License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
 with many parts BSD <http://fontforge.org/license.html>. Please read LICENSE.
 Version: 20230101
 Based on sources from 2026-01-22 20:38 UTC-ML-D-GDK3.
PythonUI_Init()
copyUIMethodsToBaseTable()
Program root: /usr

Reading mapping file
Found 601 entries

Unpacking Devicons archive

Mixing Vorillaz Devicons and possible fixes in
Found 2112 svgs
Using fixed svg for bitbucket-original.svg
Using fixed svg for akka-plain.svg
Using fixed svg for nodejs-plain.svg
Using fixed svg for angularjs-plain.svg
Using fixed svg for antdesign-plain.svg
Using fixed svg for awk-plain.svg
Using fixed svg for opera-plain.svg
Using fixed svg for bash-plain.svg
Using fixed svg for beats-plain.svg
Using fixed svg for nginx-plain.svg
Using fixed svg for yii-plain.svg
Using fixed svg for clojurescript-plain.svg
Using fixed svg for cosmosdb-plain.svg
Using fixed svg for mongodb-plain.svg
Using fixed svg for rust-original.svg
Using fixed svg for intellij-plain.svg
Using fixed svg for materializecss-plain.svg
Using fixed svg for datagrip-plain.svg
Using fixed svg for electron-plain.svg
Using fixed svg for figma-plain.svg
Using fixed svg for goland-plain.svg
Using fixed svg for googlecloud-plain.svg
Using fixed svg for graphql-plain.svg
Using fixed svg for hardhat-plain.svg
Using fixed svg for hibernate-plain.svg
Using fixed svg for homebrew-plain.svg
Using fixed svg for jetbrains-plain.svg
Using fixed svg for karma-plain.svg
Using fixed svg for kibana-plain.svg
Using fixed svg for latex-original.svg
Using fixed svg for livewire-plain.svg
Using fixed svg for llvm-plain.svg
Using fixed svg for materialui-plain.svg
Using fixed svg for matplotlib-plain.svg
Using fixed svg for maven-plain.svg
Using fixed svg for maya-plain.svg
Using fixed svg for nimble-plain.svg
Using fixed svg for norg-plain.svg
Using fixed svg for nuxtjs-plain.svg
Using fixed svg for ohmyzsh-plain.svg
Using fixed svg for okta-plain.svg
Using fixed svg for playwright-plain.svg
Using fixed svg for plotly-plain.svg
Using fixed svg for portainer-original.svg
Using fixed svg for postcss-original.svg
Using fixed svg for puppeteer-plain.svg
Using fixed svg for pytorch-original.svg
Using fixed svg for quarkus-plain.svg
Using fixed svg for qwik-plain.svg
Using fixed svg for renpy-plain.svg
Using fixed svg for sdl-plain.svg
Using fixed svg for sequelize-plain.svg
Using fixed svg for sqldeveloper-plain.svg
Using fixed svg for streamlit-plain.svg
Using fixed svg for threejs-original.svg
Using fixed svg for tomcat-plain.svg
Using fixed svg for vagrant-plain.svg
Using fixed svg for vuestorefront-plain.svg
Using fixed svg for woocommerce-plain.svg
Using fixed svg for yunohost-plain.svg
Using fixed svg for zig-plain.svg
Using fixed svg for chartjs-plain.svg
Using fixed svg for detaspace-plain.svg
Using fixed svg for filamentphp-original.svg
Using fixed svg for mapbox-plain.svg
Using fixed svg for memcached-plain.svg
Using fixed svg for newrelic-plain.svg
Using fixed svg for passport-plain.svg
Using fixed svg for proxmox-plain.svg
Using fixed svg for waku-plain.svg
Using fixed svg for web3js-plain.svg
Using fixed svg for webgpu-plain.svg
Using fixed svg for zustand-plain.svg
Generating devicons.otf with 601 glyphs
Generating GlyphInfo i_dev.sh
Finished

f:font

joeriddles committed:
Commit devicons.otf

joeriddles commented:
@all-contributors code

Finii commented on src/glyphs/devicons/mapping:
```diff
@@ -51,7 +51,7 @@ E62D E72D vorillaz/smashing_magazine.svg smashing_magazine
 E62F E72F apachespark/apachespark-original.svg apachespark
 E630 E730 apl/apl-plain.svg apl
-E631 E731 appwrite/appwrite-plain.svg appwrite
+E631 E731 appwrite/appwrite-original.svg appwrite
```
There are more such changes, substituting `original` for `plain`.

Did you do that on purpose, of came it out of the analyzer script? :thinking: 

Thanks for the PR, btw. :green_heart:

joeriddles replied:
> Did you do that on purpose, of came it out of the analyzer script

This was on purpose as some `plain` versions were removed/renamed in v2.17.0. For example, `appwrite` was renamed to `original` here https://github.com/devicons/devicon/pull/2196/files#diff-8fb0089a81b777eae04d6ac8f37becd0a07c41dd808f17e9e90eafb7411a8f60

Finii replied:
Ah I see, thanks!

It is unlikely that I have time for an in-depth look before August, sorry.
But your PR is very welcome! It should not take long to review, maybe I can squeeze it in somehow.
At the moment I work from 6 am to 8 pm every day :-(

Pietrucci-Blacher commented:
Hey @Finii, any update on the status of this PR ?

Finii committed:
Devicons: Restore manual fixes

[why]
Some icons got unfixed glyphs from the devicon svgs.

We manually created new 'plain' versions for some icons with commit
    b496b3b  Devicons: Manually optimize many icons
    6ec0645  Devicons: Fix more icons

These vanished (were no longer used) when the mappings file was
recreated.

[how]
Reinstantiate some of the handmade plain icons.

Keep old and fixed nginx-plain, they dropped plain.

Signed-off-by: Fini Jastrow <ulf.fini.jastrow@desy.de>

Finii committed:
Devicons: Fix obviously broken icons

As usual ...
And create detaspace-plain.

Signed-off-by: Fini Jastrow <ulf.fini.jastrow@desy.de>

Finii committed:
Devicons: Reduce number of points of some icons

[why]
Some icons have very much detail, so much that it can never be seen at
icon-font sizes.

[how]
Inkscape simplify.

Just three icons, before and after size comparison:

-rw-rw-r-- 1 fini fini 1273976 Jul 17 20:56 devicons.otf
-rw-rw-r-- 1 fini fini 1194732 Jul 18 08:32 devicons.otf

Signed-off-by: Fini Jastrow <ulf.fini.jastrow@desy.de>
